cogs/gaminator.py

```diff
@@ -0,0 +1,207 @@
+
+class Gaminator(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.command(name='other-games')
+    @commands.cooldown(1, 1800, commands.BucketType.user)
+    async def other_games(self, ctx):
+        logger.info(f'{ctx.message.author} called other_games')
+
+        formatted_roles_dict = format_roles_dict(ROLES['other_games']) # Key is raw role name, value is qualified name
+        pages = page_dict_as_lines(formatted_roles_dict) # Each key in formatted_roles_dict is one 'line'
+        num_pages = len(pages)
+
+        current_page_num = 0
+        current_page_dict = get_current_page_dict(pages[current_page_num]) # Gets a dict of roles indexed by emoji
+
+        menu_embed = default_embed(title=f'Other Games') # Dummy embed to update within functions
+        menu_msg = await init_menu(ctx, current_page_dict, menu_embed, num_pages)
+
+        # Internal state keeping
+        roles_to_add = []
+        roles_to_remove = []
+
+        def check_in_ctx(reaction, user):
+            return reaction.message.id == menu_msg.id
+
+        try:
+            while True:
+                reaction, user = await self.bot.wait_for('reaction_add', timeout=60.0, check=check_in_ctx)
+
+                if user == ctx.message.author: # Respond to reactions made by the command caller
+
+                    if reaction.emoji == LEFT_ARROW or reaction.emoji == RIGHT_ARROW:
+
+                        await reaction.remove(user)
+
+                        i = -1 if reaction.emoji == LEFT_ARROW else 1
+                        current_page_num = (current_page_num + i) % num_pages
+                        current_page_dict = get_current_page_dict(pages[current_page_num])
+                        await set_page(ctx, current_page_dict, menu_embed, num_pages, current_page_num, menu_msg)
+
+                    elif reaction.emoji == CHECK_MARK:
+
+                        if len(roles_to_add) > 0 or len(roles_to_remove) > 0:
+                            autorole = self.bot.get_cog('Autorole')
+                            if len(roles_to_add) > 0:
+                                await autorole.other_game_add(ctx, *[role['qualified-name'] for role in roles_to_add])
+                            if len(roles_to_remove) > 0:
+                                await autorole.other_game_remove(ctx, *[role['qualified-name'] for role in roles_to_remove])
+
+                        break
+
+                    elif reaction.emoji in [e for e in ri_alphabet(len(current_page_dict))]: # If reaction is a valid and applicable regional indicator
+
+                        await reaction.remove(user)
+
+                        if current_page_dict[reaction.emoji]['role-name'] not in [role.name for role in user.roles]: # If user doesn't already have role
+                            update_list, field_index = await update_roles_to_add(reaction, user, current_page_dict, roles_to_add) # field_index: 0 is 'Adding' field; 1 is 'Removing' field
```

Comment:
Re: https://github.com/Ghost-Proxy/Voluspa/pull/52/files#r439759071
Given that here the context for each call is known, could the `field_index` be set based on the conditional context?